pkg/k8s/watchers/pod.go

```diff
@@ -728,6 +733,28 @@ func (k *K8sWatcher) deleteHostPortMapping(pod *slim_corev1.Pod, podIPs []string
 	return nil
 }
 
+func findServiceByL3n4Addr(l3n4Addr loadbalancer.L3n4Addr, svcs []*loadbalancer.SVC) *loadbalancer.SVC {
+	for _, svc := range svcs {
+		if l3n4Addr.DeepEqual(&svc.Frontend.L3n4Addr) {
+			return svc
+		}
+	}
+	return nil
+}
+
+func deepEqualBackends(backends1 []*loadbalancer.Backend, backends2 []*loadbalancer.Backend) bool {
+	if len(backends1) != len(backends2) {
+		return false
+	}
+	for idx, backend1 := range backends1 {
```

Comment:
are we sure these will always be in the same order?

Reply:
Good point, i'm not sure if we can assume they will be in the same order. Updated the deepEquals logic to be order agnostic. PTAL